### PR TITLE
gh-96288: Add a sentence to `os.mkdir`'s docstring.

### PR DESCRIPTION
**Pull request by hagai-helman**
A clarification about the use of umask (copied from the `os` module documentation) is added to the docstring of `os.mkdir`.

The behavior described may be unexpected for some (for example, this is not the behavior of GNU's `mkdir` command - when a mode is explicitly given to it, it overrides the umask). When one is unaware of this behavior, it can lead to bugs that are difficult to notice or trace. For these two reasons, I suggest it is mentioned in the docstring.

* Issue: gh-96288

### CODE REVIEW

File: Modules/clinic/posixmodule.c.h [hunk 1]
```diff
@@ -1849,7 +1849,8 @@ PyDoc_STRVAR(os_mkdir__doc__,
 "dir_fd may not be implemented on your platform.\n"
 "  If it is unavailable, using it will raise a NotImplementedError.\n"
 "\n"
-"The mode argument is ignored on Windows.");
+"The mode argument is ignored on Windows. Where it is used, the current umask\n"
```
Comment (mdboom) [inline Modules/clinic/posixmodule.c.h:1852]: This is editing the generated output of the "argument clinic" program.  What you need to do is edit the [source file here](https://github.com/python/cpython/blob/main/Modules/posixmodule.c#L4524), run `make clinic` to regenerate the output, and commit both changes to your PR.

**Review by mdboom — CHANGES_REQUESTED** [review]
This is editing the docstring in generated output.

File: Modules/clinic/posixmodule.c.h (hunk 1 above)
Comment (hagai-helman) [reply Modules/clinic/posixmodule.c.h:1852]: Oh, thank you very much! Done.

**Comment by JelleZijlstra** [comment]
The change looks good (sorry for the wait!) but there's a merge conflict in the AC output. Could you merge main in and re-generate the AC output file?

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**